src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -15,10 +15,47 @@
 namespace System
 {
     /// <summary>
-    /// Extension methods and non-generic helpers for Span and ReadOnlySpan
+    /// Extension methods and non-generic helpers for Span, ReadOnlySpan, Memory, and ReadOnlyMemory.
     /// </summary>
     public static class Span
     {
+        /// <summary>Creates a new <see cref="ReadOnlyMemory{T}"/> over the portion of the target string.</summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is a null reference (Nothing in Visual Basic).</exception>
+        public static ReadOnlyMemory<char> AsReadOnlyMemory(this string text)
+        {
+            if (text == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
+            }
+
+            return new ReadOnlyMemory<char>(text, 0, text.Length);
+        }
+
+        /// <summary>Attempts to get the underlying <see cref="string"/> from a <see cref="ReadOnlyMemory{T}"/>.</summary>
+        /// <param name="readOnlyMemory">The memory that may be wrapping a <see cref="string"/> object.</param>
+        /// <param name="text">The string.</param>
+        /// <param name="start">The starting location in <paramref name="text"/>.</param>
+        /// <param name="length">The number of items in <paramref name="text"/>.</param>
+        /// <returns></returns>
+        public static bool TryGetString(this ReadOnlyMemory<char> readOnlyMemory, out string text, out int start, out int length)
```

Comment:
Was the parameter name inconsistency discussed during review?
Should it be `out int index` instead of `out int start` to be consistent with the other APIs?

Reply:
> Was the parameter name inconsistency discussed during review?

It wasn't. Which inconsistency?  These match the names used with Slice and the ctors. (That said, as you know I don't like the names, so it pained me to write these :smile:)

Reply:
Ah yes, `index` would be inconsistent, and `start` is fine :)